app/backends/workflow-orchestration.ts

```diff
@@ -517,7 +522,11 @@ export async function updateWorkItem(req: HarmonyRequest, res: Response): Promis
     );
     const allWorkItemsForStepComplete = (completedWorkItemCount == thisStep.workItemCount);
 
-    await maybeClearScrollSession(workItem.scrollID, allWorkItemsForStepComplete, status);
+    if (hits && job.numInputGranules != hits) {
```

Comment:
There's a complication with the `maxResults` parameter since the hits count will always end up being higher than numInputGranules if we've restricted it with the parameter. I think we can solve it by only checking if the number of hits is less than `job.numInputGranules`. So we'll never allow processing more granules than the initial numInputGranules (no different than today with scrolling), but we will successfully complete a request if there end up being fewer granules because some were deleted from CMR while the request was processing.

Reply:
Done.